crates/rpc-client/src/pathfinder/proofs.rs

```diff
@@ -139,6 +146,16 @@ pub fn verify_proof<H: HashFunctionType>(
                         index += 1;
                     }
                     TrieNode::Edge { child, path } => {
+                        let path_bits = path.value.to_bits_be();
+                        let relevant_path_bits = &path_bits[path_bits.len() - path.len as usize..];
+                        let key_bits_slice = &bits[index as usize..(index + path.len) as usize];
+                        if relevant_path_bits != key_bits_slice {
```

Comment:
@ftheirs here you are trying to cast a u64 to a usize.
Using as in not recommended as can panic. Specifically, if this code is run using a 32 bit cpu this will panic or truncate the value.

Please change it to correctly handle the error using `try_into` :)